### ISSUE TITLE
Updater: unclosed file object

### ISSUE THREAD

**Issue by jku** [theupdateframework/tuf#1099]
file_object  in `Updater._get_metadata_file()` is left unclosed whenever an exception is raised (when the metadata fails a check of any kind). There are loads of examples in the test suite, e.g.
```
python3 test_updater.py TestUpdater.test_3__update_metadata
```
will complain "ResourceWarning: unclosed file"

This is not very dangerous but makes it hard to find actual issues in test output (try running all `test_updater.py` tests). There may be other sources of this warning but the vast number of warnings from this one make it difficult to find them.

**Comment by joshuagl** [comment]
Seems like a good place to use [`finally`](https://docs.python.org/3.8/reference/compound_stmts.html#finally) as a clean up handler.

**Comment by joshuagl** [comment]
Fixed in #1147
